cypress/integration/ResourcesForResearchers/index.js

```diff
@@ -0,0 +1,264 @@
+
+And('the {string} box is displayed for {string}', (section, eligibleToolType) => {
+  if (eligibleToolType == "Y") {
+    cy.get('.facet__box--tool-subtypes').should('be.visible');
+  }
+});
+
+And('the {string} box displays the correct {string} checked', (section, area) => {
+  const regex = new RegExp(area);
+  cy.get('.r4r-DEFAULT.facet__box> h4').eq[2].should('contain', section);;
+  cy.get('.r4r-DEFAULT.facet__box >label[role="checkbox"]>input[checked]').should('be.visible');
+
+});
+And('the {string} box is displayed', (section) => {
+  cy.get('.r4r-DEFAULT.results__facets >div').find('h4').should('contain.text', section);
+
+});
+
+When('user types {string} in the search bar', (searchText) => {
+  cy.get('.searchbar__container input').type(searchText);
+});
+
+When('user clicks on the magnifying glass icon', () => {
+  cy.get('.searchbar__container button').click();
+});
+
+And("search box displays the term {string}", (searchText) => {
+  cy.get('.searchbar__container input').should('have.attr', 'value', searchText);
+});
+
+When('user hits {string} key', (enterKey) => {
+  cy.get('.searchbar__container input').type('{enter}{enter}');
+  cy.wait(2000);
+});
+
+When('user clicks on {string} button', (button) => {
+  cy.get('.r4r-DEFAULT.results__search-container > .r4r-DEFAULT.results__filter-button').click();
+});
+
+And('the search result is displayed', () => {
+  cy.get('[aria-label="search results"]').should('be.visible');
+});
+
+And('the search result title is {string} with href {string}', (title, href) => {
+  cy.get('[aria-label="search results"] > article>h2').first().should('be.visible').find('a').should('have.attr', 'href', href).should('have.text', title);
+});
+
+And('the search result description is displayed', () => {
+  cy.get('[aria-label="search results"] > article>p').first().should('be.visible');
+});
+
+And('the message is displayed as {string}', (message) => {
+  cy.get('.results__noresults').should('contain.text', message);
+});
+
+And("{string} is a link with href {string}", (linkText, linkHref) => {
+  cy.get('.results__noresults > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+Then('the resource title is displayed as {string}', (resourceTitle) => {
+  cy.get('.r4r-resource__header h1').should('contain.text', resourceTitle);
+});
+
+And("{string} is displayed with href {string}", (linkText, linkHref) => {
+  cy.get('.r4r-DEFAULT.resource__home > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+// And('the {string} is displayed', () => {
+//   cy.get('[aria-label="Resource description"]>p').should('be.visible');
+// });
+
+And('Visit Resource link is displayed', () => {
+  cy.get('.resource__link--external').should('be.visible');
+});
+
+And('search box is displayed', () => {
+  cy.get('.searchbar__container input').should('be.visible');
+});
+
+And('{string} section is displayed', (section) => {
+  cy.get('.resource__nav h2').should('be.visible');
```

Comment:
no unused variables - section parameter should be incorporated into the code
cy.get('.resource__nav h2').contains(section).should('be.visible');